_overviews/scala3-book/why-scala-3.md

```diff
@@ -33,38 +33,31 @@ It might be hard to list every benefit of Scala, but a Top Ten list looks like t
 ## 1) FP/OOP fusion
 
 More than any other language, Scala supports a fusion of the FP and OOP paradigms.
-As Martin Odersky has stated, the essence of Scala is a fusion of functional and object-oriented programming in a typed setting:
+As Martin Odersky has stated, the essence of Scala is a fusion of functional and object-oriented programming in a typed setting, with:
 
-- Functions for the logic
+- Functions for the logic, and
 - Objects for the modularity
 
 Possibly some of the best examples of this are the classes in the standard library.
-For instance, a `List` is defined as a `class`, and a new instance is created like this:
+For instance, a `List` is defined as a class — technically it’s an abstract class — and a new instance is created like this:
 
 ```scala
-val x = List(1,2,3)
+val x = List(1, 2, 3)
 ```
 
-However, what appears to the programmer to be a simple `List` is actually built from a combination of many specialized traits.
-
-{% comment %}
-TODO: Update this. It shows the Scala 2.13 hierarchy with Scala 3 syntax.
+However, what appears to the programmer to be a simple `List` is actually built from a combination of several specialized traits:
 
 ```scala
 sealed abstract class List[+A]
-  extends AbstractSeq[A],
-    LinearSeq[A],
-    LinearSeqOps[A, List, List[A]],
-    StrictOptimizedLinearSeqOps[A, List, List[A]],
-    StrictOptimizedSeqOps[A, List, List[A]],
-    IterableFactoryDefaults[A, List],
-    DefaultSerializable
+  extends AbstractSeq[A] with LinearSeq[A] with LinearSeqOps[A, [A] =>> List[A],
+  List[A]] with StrictOptimizedLinearSeqOps[A, [A] =>> List[A], List[A]] with StrictOptimizedSeqOps[A, [A] =>> List[A],
+  List[A]] with IterableFactoryDefaults[A, [A] =>> List[A]]
+  with DefaultSerializable
```

Comment:
I might be missing some context but what is the motivation for showing this complex class definition on the page “why Scala 3”?

I personally find it a bit scary, especially because it refers to some traits that should be implementation details only (we didn’t have “transparent traits” at that time). Also, I don’t think `[A] =>> List[A]` is clearer than `List`.

Reply:
I’ll see what I can do about that. The intent was to show something of a real-world example, but that may be a little too intense.